src/components/range-inputs.tsx

```diff
@@ -0,0 +1,303 @@
+import '@isrd-isi-edu/chaise/src/assets/scss/_range-input.scss';
+import { useState, useRef } from 'react';
+import { windowRef } from '@isrd-isi-edu/chaise/src/utils/window-ref';
+import { ClearInputBtn } from '@isrd-isi-edu/chaise/src/components/clear-input-btn';
+
+
+const INTEGER_REGEXP = /^\-?\d+$/;
+
+const FLOAT_REGEXP = /^\-?(\d+)?((\.)?\d+)?$/;
+
+const TIME_FORMAT = 'YYYY-MM-DDTHH:mm';
+
+const DATE_FORMAT = 'YYYY-MM-DD';
+
+/**
+ * an object for mapping the error codes to the error messages
+ */
+const errorMsgMap: {
+    [key: string]: string;
+} = {
+    'empty': 'Please enter either From or To value',
+    'range': 'From value cannot be greater than the To value',
+    'int': 'Please enter an integer value',
+    'float': 'Please enter a decimal value',
+    'date': 'Please enter a date value in YYYY-MM-DD format',
+    'timestamp': 'Please enter a time value in YYYY-MM-DD HH:MM format'
```

Comment:
Is this for both the date and time error of timestamp inputs or just the time input? If it's just the time input, let's rename this to `time`.

Reply:
no this is the type in case we want a timestamp so we render both date n time fields. In case we only want a date field we send the type as date.